### DISCUSSION TITLE
set port from .env file

### DISCUSSION THREAD

**Discussion by kailas-insnapsys** [beyondcode/laravel-websockets#813] [Q&A]
To run the websocket server the below command
php artisan websocket:serve 
this started to default on 6001 port  

the port number can be changed by passing the port parameter value
php artisan websocket:serve --port=15001
but it's fixed port

I want to change the port number should pick from the .env file 

in .env set port is 15001 then 
php artisan websocket:serve
will run on 15001 port instead of 6001

is anyone know about this?
Any help on this is appreciated

**Comment by aceraven777** [comment]
When you open your `config/websockets.php` you can see here you can set the default port by adding a `LARAVEL_WEBSOCKETS_PORT` in your `.env` file

**Comment by kailas-insnapsys** [comment]
> When you open your `config/websockets.php` you can see here you can set the default port by adding a `LARAVEL_WEBSOCKETS_PORT` in your `.env` file

I've tried that as well
when I run **php artisan websocket:serve** it's not taking the port which I've set in **config/websockets.php**

**Comment by aceraven777** [comment]
Upon checking the source code `vendor/beyondcode/laravel-websockets/src/Console/StartWebSocketServer.php`, its sad to say that the default port `6001` is hardcoded. you really have to explicitly put it on the artisan command.
```
php artisan websocket:serve --port=15001
```

**Comment by kailas-insnapsys** [comment]
> Upon checking the source code `vendor/beyondcode/laravel-websockets/src/Console/StartWebSocketServer.php`, its sad to say that the default port `6001` is hardcoded. you really have to explicitly put it on the artisan command.
> 
> ```
> php artisan websocket:serve --port=15001
> ```

so there is no way to set port number dynamically. except for this php artisan websocket:serve --port=15001 to pass the port number hardcoded
